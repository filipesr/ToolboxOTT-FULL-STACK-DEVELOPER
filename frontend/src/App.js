import { useState, useEffect } from "react";

import { Container, Button, Spinner } from "react-bootstrap";

import { GetList, GetData } from "./service/api";

import Header from "./components/Header";

import FileList from "./components/FileList";
import FileInfo from "./components/FileInfo";

function App({ initialListFiles = false, initialFileData = false }) {
  const [listFiles, setListFiles] = useState(initialListFiles);
  const [fileData, setFileData] = useState(initialFileData);
  const { file = false } = fileData;

  useEffect(() => {
    GetList(setListFiles);
  }, []);

  useEffect(() => {
    if (!file && fileData) GetData(fileData, setFileData);
  }, [file, fileData]);

  const title = file ? `File info: ${file}` : "React Test App - List Files";

  return (
    <Container fluid>
      <Header title={title} />
      <Container>
        {!listFiles && (
          <Button variant="secondary" disabled>
            <Spinner
              as="span"
              animation="grow"
              size="sm"
              role="status"
              aria-hidden="true"
            />
            Loading...
          </Button>
        )}

        {listFiles && <FileList setFile={setFileData} listFiles={listFiles} />}

        {fileData && !file && (
          <Button variant="secondary" disabled>
            <Spinner
              as="span"
              animation="grow"
              size="sm"
              role="status"
              aria-hidden="true"
            />
            Loading...
          </Button>
        )}
        {file && <FileInfo fileData={fileData} />}
      </Container>
    </Container>
  );
}

export default App;
