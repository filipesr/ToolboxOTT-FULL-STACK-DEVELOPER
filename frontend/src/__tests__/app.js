import React from "react";
import App from "../App";
import { create } from "react-test-renderer";

describe("Init App", () => {
  test("testing init app", () => {
    let tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  const initialListFiles = [];
  const initialFileName = "teste";
  test("testing init app with Data 01", () => {
    let tree = create(
      <App
        initialListFiles={initialListFiles}
        initialFileData={initialFileName}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  const initialFileData = { file: "teste", lines: [] };
  test("testing init app with Data 02", () => {
    let tree = create(
      <App
        initialListFiles={initialListFiles}
        initialFileData={initialFileData}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
