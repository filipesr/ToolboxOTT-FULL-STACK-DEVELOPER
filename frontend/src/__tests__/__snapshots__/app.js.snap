// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init App testing init app 1`] = `
<div
  className="container-fluid"
>
  <h4
    style={
      Object {
        "backgroundColor": "#f66",
        "color": "white",
        "padding": "1rem",
      }
    }
  >
    React Test App - List Files
  </h4>
  <div
    className="container"
  >
    <button
      className="btn btn-secondary"
      disabled={true}
      type="button"
    >
      <span
        aria-hidden="true"
        className="spinner-grow spinner-grow-sm"
        role="status"
      />
      Loading...
    </button>
  </div>
</div>
`;

exports[`Init App testing init app with Data 01 1`] = `
<div
  className="container-fluid"
>
  <h4
    style={
      Object {
        "backgroundColor": "#f66",
        "color": "white",
        "padding": "1rem",
      }
    }
  >
    React Test App - List Files
  </h4>
  <div
    className="container"
  >
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            File Name
          </th>
        </tr>
      </thead>
      <tbody
        style={
          Object {
            "cursor": "pointer",
          }
        }
      />
    </table>
    <button
      className="btn btn-secondary"
      disabled={true}
      type="button"
    >
      <span
        aria-hidden="true"
        className="spinner-grow spinner-grow-sm"
        role="status"
      />
      Loading...
    </button>
  </div>
</div>
`;

exports[`Init App testing init app with Data 02 1`] = `
<div
  className="container-fluid"
>
  <h4
    style={
      Object {
        "backgroundColor": "#f66",
        "color": "white",
        "padding": "1rem",
      }
    }
  >
    File info: teste
  </h4>
  <div
    className="container"
  >
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            File Name
          </th>
        </tr>
      </thead>
      <tbody
        style={
          Object {
            "cursor": "pointer",
          }
        }
      />
    </table>
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            File Name
          </th>
          <th>
            Text
          </th>
          <th>
            Number
          </th>
          <th>
            Hex
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
